src/components/CustomCheckbox.tsx

```diff
@@ -7,57 +7,70 @@ import {
   IndeterminateHighlightStar,
   NoHighlightStar,
 } from './Icons';
-import { CheckboxDisplayOptions } from '../types';
+import {
+  CheckboxDisplayOptionsHide,
+  CheckboxDisplayOptionsHighlight,
+  SelectionType,
+} from '../types';
 
 interface CustomCheckboxProps {
-  checkboxType: 'hide' | 'highlight';
+  selectionType: SelectionType;
   status: CheckboxState;
   clickHandler: () => void;
 }
 
 const CustomCheckbox: React.FunctionComponent<CustomCheckboxProps> = (
   props: CustomCheckboxProps
 ): JSX.Element => {
-  const { checkboxType, status, clickHandler } = props;
+  const { selectionType, status, clickHandler } = props;
 
-  const getDisplayOptions = (): CheckboxDisplayOptions => {
+  const getDisplayOptionsHighlight = (): CheckboxDisplayOptionsHighlight => {
     switch (status) {
       case CheckboxState.Checked:
         return {
-          hideTooltipText: 'Hide',
-          highlightTooltipText: 'Highlight',
+          highlightTooltipText: 'Remove highlight',
           highlightAriaLabel: 'true',
           highlightIcon: HighlightStar,
         };
       case CheckboxState.Unchecked:
         return {
-          hideTooltipText: 'Show',
-          highlightTooltipText: 'Remove highlight',
+          highlightTooltipText: 'Highlight',
           highlightAriaLabel: 'false',
           highlightIcon: NoHighlightStar,
         };
       case CheckboxState.Indeterminate:
         return {
-          hideTooltipText: 'Show',
-          highlightTooltipText: 'Remove highlight',
+          highlightTooltipText: 'Highlight',
           highlightAriaLabel: 'mixed',
           highlightIcon: IndeterminateHighlightStar,
         };
```

Comment:
why not just have a Star/HighlightComponent that takes in a state and then knows how to render itself?